### DISCUSSION TITLE
Pass values to hooks

### DISCUSSION THREAD

**Discussion by ghost** [sveltejs/kit#7181] [General]
How/were can you pass values to hooks?

Using this hooks to set the html lang:

```
export function handle({event,resolve,page}){
return resolve(event,{
transformPageChunk:({html})=>html.replace('%lang%',xxx)
});
}
```

Can I pass xxx from the server endpoint? Or otherwise set it based on current domain?

Thanks

**Comment by teemingc** [comment]
You can pass data through a cookie, in the URL, or a request Header.
https://kit.svelte.dev/docs/types#sveltejs-kit-requestevent
E.g.,
1. `event.cookies.get('cookie-name')`

2. `event.url.href` or `event.url.pathname`
e.g. `if (event.url.pathname.startsWith('/en/')) // then replace lang`
https://developer.mozilla.org/en-US/docs/Web/API/URL

3. `event.request.headers`
https://developer.mozilla.org/en-US/docs/Web/API/Request/headers

**Reply by ghost** [reply]
Thank you!
